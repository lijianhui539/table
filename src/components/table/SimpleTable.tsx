import { defineComponent, ref } from "vue";
import type { TableProps } from "./types";
import Pagination from "../pagination";

export default defineComponent({
  name: "SimpleTable",
  props: {
    dataSource: {
      type: Array,
      default: () => [],
    },
    columns: {
      type: Array,
      default: () => [],
    },
    rowKey: {
      type: Function,
      default: () => {
        return () => {};
      },
    },
    pageSize: {
      type: Number,
      default: 10,
    },
  },
  components: {
    Pagination,
  },
  setup(props: TableProps, { attrs, emit, slots }) {
 
    let current = ref(1);
    let onPageChange = (pageNumber: number) => {
      current.value = pageNumber
    };
    
    let renderList = props.columns.map(item => {})

    return () => {
      return (
        <>
          <table class="is-bordered is-hoverable is-fullwidth table">
            <thead>
              <tr>
                {props.columns.map((item) => {
                  return (
                    <td
                      key={item.key}
                      class="icon-shangjiantou"
                      title={item.title}
                    >
                      {item.title}
                    </td>
                  );
                })}
              </tr>
            </thead>
            <tbody>
              {props.dataSource.map((row) => {
                return (
                  <tr key={props.rowKey(row)}>
                    {props.columns.map((cell) => {
                      return cell.render ? (
                        <td key={cell.key}>{cell.render(row)}</td>
                      ) : (
                        <td key={cell.key} title={row[cell.key]}>
                          {row[cell.key]}
                        </td>
                      );
                    })}
                  </tr>
                );
              })}
            </tbody>
          </table>
          <Pagination
            total={props.dataSource.length}
            current={current.value}
            pageSize={props.pageSize}
            onChange={(val) => onPageChange(val)}
          ></Pagination>
        </>
      );
    };
  },
});
