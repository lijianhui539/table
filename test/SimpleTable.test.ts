import { shallowMount } from "@vue/test-utils";
import { expect, test } from "vitest";
import SimpleTable from "../src/components/SimpleTable";

test("mount component", () => {
  const wrapper = shallowMount(SimpleTable, {
    props: {},
  });

  expect(wrapper.html()).toMatchSnapshot();
});
