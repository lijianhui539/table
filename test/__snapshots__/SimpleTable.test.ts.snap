// Vitest Snapshot v1

exports[`mount component 1`] = `
"<table>
  <thead>
    <th>
    <td></td>
    </th>
  </thead>
  <tbody>
    <tr>
      <td>hello table</td>
    </tr>
  </tbody>
</table>"
`;
